src/compiler/checker.ts

```diff
@@ -2666,7 +2635,24 @@ module ts {
             }
         }
 
+        function getDeclaredPropertiesOfClassOrInterface(type: InterfaceType): Symbol[] {
+            resolveDeclaredMembers(type);
+            return type.declaredProperties;
+        }
+
+        function resolveDeclaredMembers(type: InterfaceType) {
+            if (!type.declaredProperties) {
+                var symbol = type.symbol;
+                type.declaredProperties = getNamedMembers(symbol.members);
+                type.declaredCallSignatures = getSignaturesOfSymbol(symbol.members["__call"]);
+                type.declaredConstructSignatures = getSignaturesOfSymbol(symbol.members["__new"]);
+                type.declaredStringIndexType = getIndexTypeOfSymbol(symbol, IndexKind.String);
+                type.declaredNumberIndexType = getIndexTypeOfSymbol(symbol, IndexKind.Number);
+            }
+        }
+
         function resolveClassOrInterfaceMembers(type: InterfaceType): void {
+            resolveDeclaredMembers(type);

```

Comment:
How will people remember to call this in the right places? There are no static hints if you forget to call it.